pkg/generator/generator.go

```diff
@@ -68,9 +72,13 @@ func New(config *Config) (*Generator, error) {
 	if config.DecryptTraverser == nil {
 		return nil, microerror.Maskf(invalidConfigError, "%T.DecryptTraverser must not be empty", config)
 	}
+	if config.ProjectVersion == "" {
+		config.ProjectVersion = project.AppControlPlaneVersion()
```

Comment:
This should be either `project.Version()` or `project.AppControlPlaneVersion()` depending on `config.UniqueApp` setting.

Reply:
It is, just higher up the chain. I did not want to play with that distinction for the sake of CLI generation. So if you don't provide a version, it picks `0.0.0`. See: https://github.com/giantswarm/config-controller/blob/handlers/service/controller/app.go#L84

Reply:
Then empty string should return invalidConfigError IMO. Otherwise it gets confusing (I just got confused).